*Replace this text with your own name and a brief description of what this notebook does.*

## Assignment 24-1: Survival of first and third class passengers

This assignment uses the Titanic training data from Lab 24.  You can find information about the different columns in the dataset [here](https://www.kaggle.com/c/titanic/data) (scroll down in the Data Description box at the top).

Estimate: <br>
a) the probability that a first class passenger survived<br>
b) the probabiltiy that a third class passenger survived

In [1]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns
import statsmodels.formula.api as smf
%matplotlib inline

train = pd.read_csv("../../data/train.csv")
train.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [2]:
first_class_filter = train["Pclass"] == 1
first_class = train[first_class_filter]
first_class.head()

PassengerId  Survived  Pclass  \
1             2         1       1   
3             4         1       1   
6             7         0       1   
11           12         1       1   
23           24         1       1   

                                                 Name     Sex   Age  SibSp  \
1   Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
3        Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
6                             McCarthy, Mr. Timothy J    male  54.0      0   
11                           Bonnell, Miss. Elizabeth  female  58.0      0   
23                       Sloper, Mr. William Thompson    male  28.0      0   

    Parch    Ticket     Fare Cabin Embarked  
1       0  PC 17599  71.2833   C85        C  
3       0    113803  53.1000  C123        S  
6       0     17463  51.8625   E46        S  
11      0    113783  26.5500  C103        S  
23      0    113788  35.5000    A6        S

In [3]:
num_first_class = first_class.shape[0]
num_first_class

216

In [4]:
survived_filter = first_class["Survived"] == 1
num_survived_first_class = first_class[survived_filter].shape[0]
num_survived_first_class

136

In [5]:
# probability a first class passenger survived
num_survived_first_class/num_first_class

0.6296296296296297

In [6]:
third_class_filter = train["Pclass"] == 3
third_class = train[third_class_filter]
third_class.head()

PassengerId  Survived  Pclass                            Name     Sex  \
0            1         0       3         Braund, Mr. Owen Harris    male   
2            3         1       3          Heikkinen, Miss. Laina  female   
4            5         0       3        Allen, Mr. William Henry    male   
5            6         0       3                Moran, Mr. James    male   
7            8         0       3  Palsson, Master. Gosta Leonard    male   

    Age  SibSp  Parch            Ticket     Fare Cabin Embarked  
0  22.0      1      0         A/5 21171   7.2500   NaN        S  
2  26.0      0      0  STON/O2. 3101282   7.9250   NaN        S  
4  35.0      0      0            373450   8.0500   NaN        S  
5   NaN      0      0            330877   8.4583   NaN        Q  
7   2.0      3      1            349909  21.0750   NaN        S

In [7]:
num_third_class = third_class.shape[0]
num_third_class

491

In [8]:
survived_filter = third_class["Survived"] == 1
num_survived_third_class = survived_filter.sum()
num_survived_third_class

119

In [9]:
# probability a third class passenger survived
num_survived_third_class/num_third_class

0.24236252545824846

1. What is the probability that a first class passenger survived?
2. What is the probability that a third class passenger survived?
3. Who is more likely to survive: a first or a third class passenger?

Write your answers below.

1. 0.630 or 63.0%
2. 0.242 or 24.2%
3. A first class passenger was more likely to survive.